### PR TITLE
My Code Review Suggestions

### PR DESCRIPTION
**Pull request by AustinGarrison**
Analysis:
  • I think you have a cool idea for your project. permissions, camera, photo access was implemented and operational. I look forward to looking at your finished project.
  • Project was available on time..
  • Program is fairly well commented,  could use a little more.
  • Program compiled on my emulator
  • A few suggestions from https://kotlinlang.org/docs/coding-conventions.html#choose-good-names.
I used this site to base my suggestions on.

Things I learned.
  • Properly implementing camera permissions and usage within the app. this is something my team was having issues with.
  • Used plenty of Toasts, I think it very unlikely a user will encounter an error your team is not prepared for. This is something I will suggest to my team
  • I'm mainly working on UI for my team, I think your main_fragment.xml is well organized and optimized for multiple different screen sizes.

### CODE REVIEW

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 1]
```diff
@@ -42,16 +42,12 @@ class MainFragment : Fragment() {
     private lateinit var currentPhotoPath: String
     lateinit var navController: NavController
 
-    protected var puzzleURI : Uri? = null
+    protected var puzzleUri : Uri? = null
```
Comment (AustinGarrison) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:45]: Coding conventions suggested acronyms follow camelCase unless the acronym is only two letters
puzzleURI --->  puzzleUri
randomURL ---> randomUrl

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -130,13 +125,12 @@ class MainFragment : Fragment() {
                 if(grantResults.isNotEmpty() && grantResults[0] == PackageManager.PERMISSION_GRANTED){
                     takePhoto()
                 } else {
-                    Toast.makeText(context, "Unable to take photo without permission", Toast.LENGTH_LONG).show()
+                    Toast.makeText(context, "Unable to take photo without permission, goto Apps & notifications within android settings to enable", Toast.LENGTH_LONG).show()
```
Comment (AustinGarrison) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:128]: My own suggestion just having this question brought up a lot when it comes to app permissions
After a user denies a app permission, they must re enable it themselves within their phone settings. Not many users know this, so I just added some info to the end of the Toast when user declines camera permissions

Just a personal suggestion, not professional advice.

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt [hunk 3]
```diff
@@ -198,4 +192,8 @@ class MainFragment : Fragment() {
         _binding = null
     }
 
+    companion object {
+        fun newInstance() = MainFragment()
+    }
+
```
Comment (AustinGarrison) [inline app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:198]: https://kotlinlang.org/docs/coding-conventions.html#choose-good-names suggests companion objects be placed at the bottom of the class
I moved your companion object to the bottom of the class

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt (hunk 1 above)
Comment (discospiff) [reply app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:45]: The jury is still out on this one. Generally, either approach is acceptable when using an acronym.  Do you have a source to cite for your choice?

File: app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/puzzlepic/ui/main/MainFragment.kt:128]: I'd exercise caution on directing a user how to use the phone operating system, as that may change over time.

While you're here, though, I'd consider something that will align with best practices: externalize the string to strings.xml.

**Comment by discospiff** [comment]
I'm neutral on merging this branch (the group can decide), as some of the changes are programmer's preference.  I recommend that the group review and decide whether or not to use or abandon these changes.

**Pull request closed without merging**